capella2polarion/converters/text_work_item_provider.py

```diff
@@ -0,0 +1,125 @@
+# Copyright DB InfraGO AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+"""Provides a class to generate and inset text work items in documents."""
+import polarion_rest_api_client as polarion_api
+from lxml import html
+
+from capella2polarion.converters import polarion_html_helper as html_helper
+
+
+class TextWorkItemProvider:
+    """Class providing text work items, their generation and insertion."""
+
+    def __init__(
+        self,
+        text_work_item_id_field: str = html_helper.TEXT_WORK_ITEM_ID_FIELD,
+        text_work_item_type: str = html_helper.TEXT_WORK_ITEM_TYPE,
+        existing_text_work_items: list[polarion_api.WorkItem] | None = None,
+    ):
+        self.old_text_work_items: dict[str, polarion_api.WorkItem] = {}
+        for work_item in existing_text_work_items or []:
+            # We only use those work items which have an ID defined by us
+            if text_id := work_item.additional_attributes.get(
+                text_work_item_id_field
+            ):
+                if text_id in self.old_text_work_items:
+                    raise ValueError(
+                        f"There are multiple text work items with "
+                        f"{text_work_item_id_field} == {text_id}"
+                    )
+
+                self.old_text_work_items[text_id] = work_item
+
+        self.text_work_item_id_field = text_work_item_id_field
+        self.text_work_item_type = text_work_item_type
+        self.new_text_work_items: dict[str, polarion_api.WorkItem] = {}
+
+    def generate_text_work_items(
+        self,
+        content: list[html.HtmlElement] | str,
```

Comment:
There is a problem with this type annotation mentioned earlier. But `content: list[html.HtmlElement | str] | str` isn't allowed.
Hmm. Is `lxmlhtml.fragments_fromstring` ever returning `list[str]`? I think in capellambse we completely ignore `lxml` because of the poor type annotations there.

Reply:
same as above, pls recheck